File: v1/Demo/source/demo_set_pos_pid.cpp
#include "../include/Handapi.h"

using namespace HandProtocol;

int main()
{
    Handapi api;
    api.init(LEFT_IP);
    api.calibration();
        
    // set
    uint8_t id = 1;
    vector<float> _pid {4.0, 2.0, 0.0};
    if (api.set_pos_pid(id, _pid) == ResultCode::FAIL)
    {
        Logger::get_instance()->print_trace_error("set_pos_pid failed\n");
    }
    OS::get_instance()->delay_s(1);
    Logger::get_instance()->print_trace("set_pos_pid success");
    std::cout<<std::endl;

    // verify
    vector<float> pos_pid = api.get_pos_pid();
    if (pos_pid.size() != 18)
    {
        Logger::get_instance()->print_trace_error("get pos_pid failed\n");
        return ResultCode::FAIL;
    }
    
    std::cout<<"get pos_pid success : ";
    for (int i = 0; i < pos_pid.size(); ++i) {
        std::cout << pos_pid[i] << " ";
    }
    std::cout << std::endl;

    // reset position
    std::vector<float> _back{-200, -200, -200, -200, -200, -200}, _forward{200, 200, 200, 200, 200, 200}, _stop{0, 0, 0, 0, 0, 0};
    api.set_pwm(_back);
    usleep(2000000);
    api.set_pwm(_stop);
    usleep(10000); 
    return 0;

}

File: v1/Demo/demo_set_pos_pid.py
import sys, os
sys.path.append(os.path.join(os.path.dirname(__file__), "../"))
from Hand_Sdk.HandSocket import *
from Hand_Sdk.Handapi import *
import time, math

def main():
    ip_left = "192.168.137.39" # lefthand
    # ip_right = "192.168.137.19" # righthand
    api = API(ip_left)
    api.calibration()
    time.sleep(1)

    # set
    id = 1
    _pid = [4.0, 2.0, 0.0]
    if api.set_pos_pid(id, _pid) == FunctionResult.FAIL:
        logger.print_trace_error("set_pos_pid failed")
    else:    
        logger.print_trace("set_pos_pid success")
    time.sleep(1)

    # verify
    pos_pid = api.get_pos_pid()
    if len(pos_pid) != 18:
        logger.print_trace_error("get pos_pid failed")
    else:
        logger.print_trace("get pos_pid success", pos_pid)

    # reset position
    _back, _stop = [-200, -200, -200, -200, -200, -200], [0, 0, 0, 0, 0, 0]
    api.set_pwm(_back)
    time.sleep(2)
    api.set_pwm(_stop)
    time.sleep(0.01)

if __name__ == "__main__":
    main()
